### PR TITLE
Allow special headers w/ arbitrary name casing

### PR DESCRIPTION
**Pull request by svix-james**
Allows adding headers with arbitrary casing
so that we can have headers that are all-caps,
all upper-case, or whatever else we want.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,697 @@
+{
```
Comment (tasn) [inline .vscode/launch.json]: Remove?

File: src/proto/h1/role.rs [hunk 2]
```diff
@@ -1441,9 +1442,17 @@ fn write_headers_title_case(headers: &HeaderMap, dst: &mut Vec<u8>) {
...
 
-fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>) {
+fn write_headers<'a>(headers: &HeaderMap, dst: &mut Vec<u8>, special_headers: Option<&'a HashMap<&'static str, &'static [u8]>>) {
     for (name, value) in headers {
-        extend(dst, name.as_str().as_bytes());
+        let name = if let Some(special_headers) = special_headers.as_ref() {
+            match special_headers.get(&name.as_str()) {
+                Some(x) => x,
+                None => name.as_str().as_bytes(),
+            }
```
Comment (tasn) [inline src/proto/h1/role.rs:1451]:
```suggestion
            special_headers.get(&name.as_str()).unwrap_or_else(|| name.as_str().as_bytes())
```
Does this work?

File: src/proto/h1/role.rs [hunk 3]
```diff
@@ -1441,9 +1442,17 @@ fn write_headers_title_case(headers: &HeaderMap, dst: &mut Vec<u8>) {
     }
 }
 
-fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>) {
+fn write_headers<'a>(headers: &HeaderMap, dst: &mut Vec<u8>, special_headers: Option<&'a HashMap<&'static str, &'static [u8]>>) {
```
Comment (tasn) [inline src/proto/h1/role.rs]: I don't think we actually need the `'a`, removing it should be fine.
```suggestion
fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>, special_headers: Option<&HashMap<&'static str, &'static [u8]>>) {
```

File: src/proto/h1/role.rs [hunk 4]
```diff
@@ -2666,6 +2682,31 @@ mod tests {
...
     }
 
+    #[test]
+    fn test_write_special_headers() {
+        let special_headers = HashMap::from([
+                    ("sardine-webhook-version", "Sardine-Webhook-Version".as_bytes()),
+                    ("lob-signature-timestamp", "Lob-Signature-Timestamp".as_bytes()),
+                    ("lob-signature", "Lob-Signature".as_bytes()),
+                    ("epoch", "EPOCH".as_bytes()),
+                    ("x_stilt_hmac", "X_STILT_HMAC".as_bytes()),
```
Comment (tasn) [inline src/proto/h1/role.rs]: Use different examples.

File: src/proto/h1/role.rs (hunk 2 above)
Comment (svix-james) [reply src/proto/h1/role.rs:1451]: Yes

Comment (svix-james) [reply src/proto/h1/role.rs:1451]: No, same issue as before.

**Review by tasn — APPROVED** [review]

**Pull request merged**